### PR TITLE
Use `IEx.Pry.annotate_quoted/3` in `IEx.Pry.dbg/3`

### PR DESCRIPTION
**Pull request by whatyouhide**
This is just a refactor, no change in functionality.

### CODE REVIEW

File: lib/iex/lib/iex/pry.ex [hunk 1]
```diff
@@ -111,7 +111,7 @@ defmodule IEx.Pry do
   @doc """
   Annotate quoted expression with line-by-line `IEx.Pry` debugging steps.
 
-  It expected the `quoted` expression to annotate, a `condition` that controls
+  It expects the `quoted` expression to annotate, a boolean `condition` that controls
```
Comment (josevalim) [inline lib/iex/lib/iex/pry.ex:114]: `condition` is a quoted expression that evaluates to a boolean.

**Review by josevalim — APPROVED** [review]
LGTM! The idea is that we break the pipeline into lines and then use the regular annotate_quoted? If so, that's pretty smart!

**Comment by whatyouhide** [comment]
@josevalim yeah exactly.

This:

```elixir
12 |> :crypto.strong_rand_bytes()
|> Base.encode16()
|> String.downcase() |> String.slice(0..5)
|> dbg()
```

becomes:

```elixir
(
  value = 12 |> :crypto.strong_rand_bytes()
  print_dbg_pipe(...)
)
(
  value = Base.encode16(value)
  print_dbg_pipe(...)
)
(
  value = String.slice(String.downcase(value), 0..5)
  print_dbg_pipe(...)
)
```

Then, we feed the thing above to `annotate_quoted/3`. The trick is that `annotate_quoted/3` doesn't look at lines, it looks at exprs in a block, so we make a block where exprs are what's between parens above. It Works™! Thanks for the really kind words but I felt stupid the whole time doing this, I thought this is what you meant and I was like "man it took me like an hour to figure this out" 😄

File: lib/iex/lib/iex/pry.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/iex/lib/iex/pry.ex:114]: Yep good catch, fixed!

**Review by josevalim — APPROVED** [review]

**Pull request merged**